crates/lib/src/db/raw_def/v9.rs

```diff
@@ -34,19 +41,21 @@ pub type RawIdentifier = Box<str>;
 #[derive(Debug, Clone, Default, ser::Serialize, de::Deserialize)]
 #[cfg_attr(feature = "test", derive(PartialEq, Eq, PartialOrd, Ord))]
 pub struct RawModuleDefV9 {
-    /// The types used in the module.
+    /// The `Typespace` used by the module.
     ///
     /// `AlgebraicTypeRef`s in the table, reducer, and type alias declarations refer to this typespace.
     ///
-    /// Any `Product` or `Sum` types used transitively by the module MUST be declared in this typespace.
-    ///
-    /// Every `Product`, `Sum`, and `Ref` type in this typespace MUST have a corresponding `RawTypeDefV9` declaration in the `types` field, with a module-unique name.
+    /// The typespace must satisfy `Typespace::is_nominal_normal_form`. That is, all types referenced by this typespace must either:
+    /// 1. satisfy `AlgebraicType::is_nominal_normal_form`,
+    /// 2. or be a `Sum`/`ProductType`s whose fields satisfy the same.
     ///
-    /// All product and sum types in this typespace MUST have the [default element ordering](crate::db::default_element_ordering) UNLESS they declare a custom ordering via their `RawTypeDefV9`.
+    /// Types satisfying condition 2 are called *non-nominal types*.
```

Comment:
It's definitely a bad name because that's not what it's supposed to imply... Open to suggestions.

What I'm calling a "nominal" type is something that can be used by client codegen to generate a reference to a type. If you have a `Ref(...)`, of course, then you generate `"{type_names[ref_]}"` in the client. Similarly, if you have `Array(Ref(...))`, you generate `"Array<{type_names[ref_]}>"`, etc.
There are also some special types, which look structural, but have special reference-generating code, like `AlgebraicType::option(AlgebraicType::ref(...))` or `AlgebraicType::identity`.

What I'm calling a "non-nominal" type is something that can be used by client codegen to generate a type *definition*. That is, something in the module's `typespace` with an attached name. Non-nominal is really not the right word for this.

Reply:
Fixed in #1590